app/models/vendor.rb

```diff
@@ -0,0 +1,33 @@
+class Vendor < ActiveRecord::Base
+  has_many :products
+  belongs_to :market
+
+  def total_sales
+    total = 0
+    self.products.each do |product|
+      total += product.sales.inject(0) { |sum, sale| sum + sale.amount }
+    end
+    return total
+  end
+
+  def monthly_sales
+    month_sales = []
+
+    self.products.each do |product|
+      sales = product.sales.find_all do |sale|
+        unless sale.purchase_time.nil?
+          Time.now.month == sale.purchase_time.month && Time.now.year == sale.purchase_time.year
```

Comment:
Seems like a lot of logic happening in this one line. I would recommend splitting out `Time.now` into a local variable and possibly purchase time as well so the month/year comparison is more clear.